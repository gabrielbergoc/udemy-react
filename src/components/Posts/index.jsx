import P from 'prop-types';
import PostCard from '../PostCard';
import './styles.css';

const Posts = ({ posts }) => (
  <div className="posts">
    {posts.map((post) => (
      <PostCard key={post.id} post={post} />
    ))}
  </div>
);

Posts.propTypes = {
  posts: P.arrayOf(
    P.shape({
      id: P.number.isRequired,
      title: P.string.isRequired,
      body: P.string.isRequired,
      cover: P.string.isRequired,
    }),
  ).isRequired,
};

export default Posts;
