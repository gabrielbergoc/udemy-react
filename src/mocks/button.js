export const buttonProps = {
  text: 'button text',
  title: 'button title',
  onClick: jest.fn(),
  disabled: false,
};
