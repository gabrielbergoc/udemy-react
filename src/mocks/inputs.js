export const searchInputProps = {
  value: "a value",
  handleChange: jest.fn(),
  placeholder: "a placeholder",
}
