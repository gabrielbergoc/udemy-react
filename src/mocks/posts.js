export const posts = [
  {
    id: 1,
    title: "post 1 title",
    body: "post 1 body",
    cover: "img/img1.png",
  },
  {
    id: 2,
    title: "post 2 title",
    body: "post 2 body",
    cover: "img/img2.png",
  },
  {
    id: 3,
    title: "post 3 title",
    body: "post 3 body",
    cover: "img/img3.png",
  },
];

export const post = posts[0];
