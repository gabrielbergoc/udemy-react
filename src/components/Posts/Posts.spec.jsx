import { render, screen } from "@testing-library/react";
import Posts from ".";
import { posts } from "../../mocks/posts";


describe("<Posts />", () => {
  it("should render Posts", () => {
    render(<Posts posts={posts} />);

    const allHeadings = screen.getAllByRole("heading");
    const allBodies = screen.getAllByText(/body/i);
    const allImages = screen.getAllByRole("img");

    expect(allHeadings).toHaveLength(posts.length);
    expect(allBodies).toHaveLength(posts.length);
    expect(allImages).toHaveLength(posts.length);
  });

  it("should match snapshot", () => {
    const { container } = render(<Posts posts={posts} />);

    expect(container).toMatchSnapshot();
  });
});
