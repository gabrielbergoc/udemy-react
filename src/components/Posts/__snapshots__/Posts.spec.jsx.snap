// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="post 1 title"
        src="img/img1.png"
      />
      <div
        class="post-content"
      >
        <h2>
          post 1 title
        </h2>
        <p>
          post 1 body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="post 2 title"
        src="img/img2.png"
      />
      <div
        class="post-content"
      >
        <h2>
          post 2 title
        </h2>
        <p>
          post 2 body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="post 3 title"
        src="img/img3.png"
      />
      <div
        class="post-content"
      >
        <h2>
          post 3 title
        </h2>
        <p>
          post 3 body
        </p>
      </div>
    </div>
  </div>
</div>
`;
